src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php

```diff
@@ -231,7 +231,7 @@ public function hasMorph($relation, $types, $operator = '>=', $count = 1, $boole
         }
 
         foreach ($types as &$type) {
-            $type = Relation::getMorphedModel($type) ?? $type;
+            $type = $this->model::getActualClassNameForMorph($type) ?? $type;
```

Comment:
If the subclass does not define `getActualClassNameForMorph()`, it will still call `Model::getActualClassNameForMorph()`, Doing this allows subclasses to override `getActualClassNameForMorph()`, avoiding the dependency on `Relation::$morphMap`